language/vm/vm-runtime/src/libra_vm.rs

```diff
@@ -245,9 +245,7 @@ impl LibraVM {
                 self.run_prologue(gas_schedule, &mut ctx, &txn_data)?;
                 Ok(VerifiedTranscationPayload::Module(module.code().to_vec()))
             }
-            TransactionPayload::WriteSet(change_set) => {
-                Ok(VerifiedTranscationPayload::ChangeSet(change_set.clone()))
-            }
+            TransactionPayload::WriteSet(_) => Err(VMStatus::new(StatusCode::UNREACHABLE)),
```

Comment:
AC and Mempool is an non-existent problem right now as genesis is the only place where the code is used.

Reply:
Does it make sense to have two different WriteSets ... one as a Top level transaction and another as a UserTransaction? The fact that genesis uses this path seems... confusing. At a minimum should we add a TODO to delete this if not just delete it?

Reply:
The two WriteSet is more of a historic reason where back in the days we only have `SignedTransaction` and will have no way to represent genesis other than adding `TransactionPayload::WriteSet`. We should definitely keep one in the end but I would say which one to keep remains debatable. The main problem is if we would allow those transactions to appear in AC/mempool.